src/dmd/e2ir.d

```diff
@@ -353,6 +354,13 @@ private elem *callfunc(const ref Loc loc,
             {
                 ethis = addressElem(ec, ectype);
             }
+            if (ethis2)
```

Comment:
duplicated code maybe it could go into a function and clearify intend?

Reply:
maybe

Reply:
done